### PR TITLE
Add ROC metric

### PR DESCRIPTION
**Pull request by joanne-tsai**
Before there was no ROC metric. Having an ROC metric helps with visualizing the performance of the algorithm. This adds the ROC metric as well as some edge test cases to ensure that the ROC curve is calculated correctly.

### CODE REVIEW

File: isic_challenge_scoring/metrics.py [hunk 1]
```diff
@@ -174,3 +174,9 @@ def auc_above_sensitivity(
 def average_precision(truth_probabilities: pd.Series, prediction_probabilities: pd.Series) -> float:
     ap = sklearn.metrics.average_precision_score(truth_probabilities, prediction_probabilities)
     return ap
+
+
+def roc(truth_probabilities: pd.Series, prediction_probabilities: pd.Series):
+    fpr, tpr, thresholds = sklearn.metrics.roc_curve(truth_probabilities, prediction_probabilities)
+    roc = list(map(lambda x, y, t: {'x': x, 'y': y, 'thresh': t}, fpr, tpr, thresholds))
```
Comment (brianhelba) [inline isic_challenge_scoring/metrics.py]: Could we use keys of `tpr`, `fpr`, and `threshold` for this? X and Y are properties of the ROC rendering, rather than measurements of the algorithm.

File: isic_challenge_scoring/task3.py [hunk 2]
```diff
@@ -56,6 +56,7 @@ def compute_metrics(truth_file_stream, prediction_file_stream) -> Dict[str, Dict
             'ap': metrics.average_precision(
                 truth_category_probabilities, prediction_category_probabilities
             ),
+            # 'roc': metrics.roc(truth_category_probabilities, prediction_category_probabilities),
```
Comment (brianhelba) [inline isic_challenge_scoring/task3.py]: Why isn't this being used? Why do we need to set it in another loop below?

Comment (joanne-tsai) [reply isic_challenge_scoring/task3.py]: If I put it in that loop, the macro_average tries to calculate the average of the roc curve and throws an error. I couldn't figure out how to make the macro_average skip roc so I thought to just add another loop at the end to add the roc metric in.

**Review by brianhelba — APPROVED** [review]
Looks great. I particularly like the tests.

For large inputs, this might be outputting too many ROC points, but we can address that if necessary in the future.
